src/node_crypto.cc

```diff
@@ -2855,8 +2855,7 @@ bool CipherBase::CheckCCMMessageLength(int message_len) {
 bool CipherBase::IsAuthenticatedMode() const {
   // Check if this cipher operates in an AEAD mode that we support.
   CHECK(ctx_);
-  const int mode = EVP_CIPHER_CTX_mode(ctx_.get());
-  return IsSupportedAuthenticatedMode(mode);
+  return IsSupportedAuthenticatedMode(EVP_CIPHER_CTX_cipher(ctx_.get()));
```

Comment:
Would overloading `IsSupportedAuthenticatedMode()` to also accept the `EVP_CIPHER_CTX` type make sense? It would shorten these 3 repeated calls.